### ISSUE TITLE
[Research] OUI Compliance audit of the `vis_type_vega` plugin

### ISSUE THREAD

**Issue by BSFishy** [opensearch-project/OpenSearch-Dashboards#4103]
Tracking issue for OUI compliance audit of the `vis_type_vega` plugin.

Follow the steps in #3945 to complete the audit, then open another issue with findings. Once the work of performing the audit is done, this issue can be closed.

If this issue isn't already assigned to someone and you'd like to take on this work, please ping @BSFishy in the comments and I can assign it to you.

**Comment by miamia1999** [comment]
Could you, please, assign this issue to me? Thank you

**Comment by joshuarrrr** [comment]
@miamia1999 Go ahead and give it a go - there are likely a number of changes we'll need to make here long term, and I can help if you have any questions.
